pkg/identityserver/application_access.go

```diff
@@ -434,10 +434,24 @@ func (is *IdentityServer) deleteApplicationCollaborator(
 		return nil, err
 	}
 	err = is.store.Transact(ctx, func(ctx context.Context, st store.Store) error {
-		removedRights, err := st.GetMember(ctx, req.GetCollaboratorIds(), req.GetApplicationIds().GetEntityIdentifiers())
+		removedRights, err := st.GetMember(
+			ctx, req.GetCollaboratorIds(), req.GetApplicationIds().GetEntityIdentifiers(),
+		)
 		if err != nil {
 			return err
 		}
+		app, err := st.GetApplication(
+			ctx,
+			req.GetApplicationIds(),
+			store.FieldMask([]string{"administrative_contact", "technical_contact"}),
+		)
+		if err != nil {
+			return err
+		}
+		if app.GetAdministrativeContact().IDString() == req.GetCollaboratorIds().IDString() ||
```

Comment:
Please just use `proto.Equal` between the identifiers, instead of comparing the strings directly.